core/src/main/java/io/dekorate/SelectorDecoratorFactory.java

```diff
@@ -0,0 +1,17 @@
+
+package io.dekorate;
+
+import io.dekorate.kubernetes.decorator.NamedResourceDecorator;
+
+/**
+ * A factory for creating {@link SelectorDecorator} instances.
+ */
+public interface SelectorDecoratorFactory {
```

Comment:
This is a real nice, :+1: !

Can you however add a comment on why we need it (`Deployment` vs `DeploymentConfig`) as I am sure it won't be immediately evident when I read the code in a few months :)